packages/common/src/services/FavoriteService.ts

```diff
@@ -15,31 +25,60 @@ export default class FavoriteService {
 
   private readonly apiService;
   private readonly storageService;
+  private readonly accountService;
 
-  constructor(apiService: ApiService, storageService: StorageService) {
+  constructor(@inject(INTEGRATION_TYPE) integrationType: string, apiService: ApiService, storageService: StorageService) {
     this.apiService = apiService;
     this.storageService = storageService;
+    this.accountService = getNamedModule(AccountService, integrationType, false);
```

Comment:
Does notmal injection not work for this?

Reply:
Only for the integration services. There is no difference between injecting the accountService in a controller or another service, unfortunately.

Reply:
Maybe we should reconsider (in a future PR) how we're handling that then. I would like to minimize this boilerplate.

Reply:
I tried this, but it does change the containers a bit:

```tsx
// Functions
container.bind(INTEGRATION_TYPE).toDynamicValue(getIntegrationType);

const getDynamicIntegration =
  <T>(identifier: interfaces.ServiceIdentifier<T>) =>
  (context: interfaces.Context) => {
    const integrationType = context.container.get(INTEGRATION_TYPE) as string;
    return context.container.getNamed<T>(identifier, integrationType);
  };

// Integration getters
container.bind(AccountService).toDynamicValue(getDynamicIntegration('AccountService'));
container.bind(CheckoutService).toDynamicValue(getDynamicIntegration('CheckoutService'));
container.bind(SubscriptionService).toDynamicValue(getDynamicIntegration('SubscriptionService'));
container.bind(ProfileService).toDynamicValue(getDynamicIntegration('ProfileService'));

// Cleeng integration implementation
container.bind(DETERMINE_INTEGRATION_TYPE).toConstantValue(isCleengIntegrationType);
container.bind(CleengService).toSelf();
container.bind<AccountService>('AccountService').to(CleengAccountService).whenTargetNamed(INTEGRATION.CLEENG);
container.bind<CheckoutService>('CheckoutService').to(CleengCheckoutService).whenTargetNamed(INTEGRATION.CLEENG);
container.bind<SubscriptionService>('SubscriptionService').to(CleengSubscriptionService).whenTargetNamed(INTEGRATION.CLEENG);

// JWP integration implementation
container.bind(DETERMINE_INTEGRATION_TYPE).toConstantValue(isJwpIntegrationType);
container.bind(JWPEntitlementService).toSelf();
container.bind<AccountService>('AccountService').to(JWPAccountService).whenTargetNamed(INTEGRATION.JWP);
container.bind<CheckoutService>('CheckoutService').to(JWPCheckoutService).whenTargetNamed(INTEGRATION.JWP);
container.bind<SubscriptionService>('SubscriptionService').to(JWPSubscriptionService).whenTargetNamed(INTEGRATION.JWP);
container.bind<ProfileService>('ProfileService').to(JWPProfileService).whenTargetNamed(INTEGRATION.JWP);
```

With the above, we can return back to using inversify injection:

```tsx
export default AccountController {
  constructor(
    accountService: AccountService,
    checkoutService: CheckoutService,
    subscriptionService: SubscriptionService,
    favoritesController: FavoritesController,
    watchHistoryController: WatchHistoryController,
    profileController: ProfileController,
  ) {
    this.accountService = accountService;
    this.checkoutService = checkoutService;
    this.subscriptionService = subscriptionService;
    this.favoritesController = favoritesController;
    this.watchHistoryController = watchHistoryController;
    this.profileController = profileController;

    this.features = accountService ? this.accountService.features : DEFAULT_FEATURES;
  }
}
```

How does this work?

We only register the integration services once on the abstract classes to dynamic values. These will be resolved once requested. In each dynamic value, we request the implementation based on the integration type returned by the `INTEGRATION_TYPE` value. 

The only downside is that you must register the implementations with a different service identifier (plain string in the example above), but it will greatly simplify the usage in controllers 😄.

It essentially moves the `getNamedModule` code to the service container.